vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/test/java/com/vaadin/flow/component/checkbox/tests/CheckboxGroupTest.java

```diff
@@ -78,19 +78,14 @@ public void setValueNull_throws() {
     }
 
     @Test
-    public void setReadOnlyCheckboxGroup_groupIsReadOnlyAndDisabled() {
+    public void setReadOnlyCheckboxGroup_groupIsReadOnly() {
         CheckboxGroup<String> group = new CheckboxGroup<>();
         group.setItems("foo", "bar");
         group.setReadOnly(true);
         Assert.assertTrue(group.isReadOnly());
 
         Assert.assertEquals(Boolean.TRUE.toString(),
-                group.getElement().getProperty("disabled"));
-
-        long disabledChildCount = group.getChildren().filter(
-                child -> child.getElement().getProperty("disabled", false))
-                .count();
-        Assert.assertEquals(group.getChildren().count(), disabledChildCount);
```

Comment:
Depending on which solution you decide on above, we might want to restore checking individual checkboxes to be readonly as well.